ludwig/encoders/text_encoders.py

```diff
@@ -2374,3 +2380,99 @@ def output_shape(self) -> torch.Size:
+        # want to exceed the maximum possible context length so we also check for that.
+        # if self.encoder_config.max_sequence_length:
+        #     max_sequence_length = self.encoder_config.max_sequence_length
+        #     self.max_sequence_length = (
+        #        max_sequence_length if max_sequence_length <= self.context_len else self.context_len
+        #     )
+        # else:
+        print(f"CONTEXT LENGTH: {self.context_len}")
+        self.max_sequence_length = self.context_len
+
+        # Initialize tokenizer
+        self.tokenizer = HFTokenizer(self.config.base_model).tokenizer
+
+        self.attention_masks = None
+
+        clear_data_cache()
+
+    @staticmethod
+    def get_schema_cls() -> Type[BaseEncoderConfig]:
+        return LLMEncoderConfig
+
+    @property
+    def input_shape(self) -> torch.Size:
+        return torch.Size([self.max_sequence_length])
+
+    @property
+    def output_shape(self) -> torch.Size:
+        return torch.Size([self.max_sequence_length, self.model_config.hidden_size])
+
+    def get_embedding_layer(self) -> nn.Module:
+        return self
+
+    def initialize_adapter(self):
+        """If an adapter config is provided, we want to wrap the model with a PEFT model for fine-tuning."""
+        if self.config.adapter:
+            self.model = initialize_adapter(self.model, self.config)
+
+            logger.info("==================================================")
+            logger.info("Trainable Parameter Summary For LLM Encoder Fine-Tuning")
+            logger.info(f"Fine-tuning with adapter: {self.config.adapter.type}")
+            self.model.print_trainable_parameters()
+            logger.info("==================================================")
+
+    def prepare_for_training(self):
+        # TODO: this implementation will not work if resuming from a previous checkpoint. Need to fix this.
+        if self.config.quantization:
+            self.prepare_for_quantized_training()
+        self.initialize_adapter()
+
+    def prepare_for_quantized_training(self):
+        from peft import prepare_model_for_kbit_training
+
+        self.model = prepare_model_for_kbit_training(self.model, use_gradient_checkpointing=False)
+
+    def forward(self, inputs: torch.Tensor, mask: Optional[torch.Tensor] = None):
+        # Wrap with flash attention backend for faster generation
+        with torch.backends.cuda.sdp_kernel(enable_flash=True, enable_math=False, enable_mem_efficient=False) if (
+            torch.cuda.is_available() and self.curr_device.type == "cuda"
+        ) else contextlib.nullcontext():
+            # TODO (jeffkinnison): Determine why the 8-bit `SCB` and `CB` matrices are deleted in the forward pass
+            model_outputs = self.model(input_ids=inputs, return_dict=True, output_hidden_states=True).hidden_states[-1]
+        return {ENCODER_OUTPUT: model_outputs}
+
+    def _save_to_state_dict(self, destination: Dict, prefix: str, keep_vars: bool):
```

Comment:
OOC, could you drop a code pointer here for where this is called? Would like to understand where this `destination` dict is coming from!